### PR TITLE
KAFKA-10447: Migrate tools module to JUnit 5 and mockito

### PR DESCRIPTION
**Pull request by ijuma**
This change sets the groundwork for migrating other modules incrementally.

Main changes:
- Replace `junit` 4.13 with `junit-jupiter` and `junit-vintage` 5.7.0-RC1.
- All modules except for `tools` depend on `junit-vintage`.
- `tools` depends on `junit-jupiter`.
- Convert `tools` tests to JUnit 5.
- Update `PushHttpMetricsReporterTest` to use `mockito` instead of `powermock` and `easymock`
(powermock doesn't seem to work well with JUnit 5 and we don't need it since mockito can mock
static methods).
- Update `mockito` to 3.5.7.
- Update `TestUtils` to use JUnit 5 assertions since `tools` depends on it.

Unrelated clean-ups:
- Remove `unit` from package names in a few `core` tests.
- Replace `try/catch/fail` with `assertThrows` in a number of places.
- Tag `CoordinatorTest` as integration test.
- Remove unnecessary type parameters when invoking methods and constructors.

Tested with IntelliJ and gradle. Verified that the following commands work as expected:
* ./gradlew tools:unitTest
* ./gradlew tools:integrationTest
* ./gradlew tools:test
* ./gradlew core:unitTest
* ./gradlew core:integrationTest
* ./gradlew clients:test

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: tools/src/test/java/org/apache/kafka/trogdor/coordinator/CoordinatorTest.java [hunk 1]
```diff
@@ -62,17 +62,18 @@
...
-import static org.junit.Assert.assertFalse;
-import static org.junit.Assert.fail;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
 
+@Tag("integration")
```
Comment (ijuma) [inline tools/src/test/java/org/apache/kafka/trogdor/coordinator/CoordinatorTest.java:72]: This test is much slower than the rest in `tools`, so I think it makes sense to mark it as `integration`.

**Comment by ijuma** [comment]
Unrelated flaky failures:

```
Build / JDK 15 / kafka.network.DynamicConnectionQuotaTest.testDynamicListenerConnectionCreationRateQuota
Build / JDK 8 / kafka.api.PlaintextAdminIntegrationTest.testConsumerGroups
Build / JDK 8 / org.apache.kafka.streams.integration.EosBetaUpgradeIntegrationTest.shouldUpgradeFromEosAlphaToEosBeta[true]

```

**Review by omkreddy — APPROVED** [review]
@ijuma Thanks for the PR. LGTM.

**Comment by ijuma** [comment]
Rebased to fix a trivial conflict. JDK 11 and JDK 8 builds passed, JDK 15 failed with 2 unrelated flaky test failures:

> kafka.network.DynamicConnectionQuotaTest.testDynamicListenerConnectionCreationRateQuota
> kafka.network.DynamicConnectionQuotaTest.testDynamicListenerConnectionCreationRateQuota
> kafka.server.DynamicBrokerConfigTest.testPasswordConfigEncoderSecretChange

**Pull request merged**